MultimodalQnA/ui/gradio/multimodalqna_ui_gradio.py

```diff
@@ -315,6 +315,64 @@ def ingest_image_gen_caption(filepath, request: gr.Request):
     return
 
 
+def ingest_image_with_text(filepath, text, request: gr.Request):
+    yield (gr.Textbox(visible=True, value="Please wait for your uploaded image to be ingested into the database..."))
+    verified_filepath = os.path.normpath(filepath)
+    if not verified_filepath.startswith(tmp_upload_folder):
+        print("Found malicious image file name!")
+        yield (
+            gr.Textbox(
+                visible=True,
+                value="Your uploaded image's file name has special characters that are not allowed. Please consider updating the file name!",
+            )
+        )
+        return
+    basename = os.path.basename(verified_filepath)
+    dest = os.path.join(static_dir, basename)
+    shutil.copy(verified_filepath, dest)
+    text_basename = "{}.txt".format(os.path.splitext(basename)[0])
+    text_dest = os.path.join(static_dir, text_basename)
+    with open(text_dest, "a") as file:
+        file.write(text)
+    print("Done copying uploaded files to static folder!")
+    headers = {
+        # 'Content-Type': 'multipart/form-data'
+    }
+    files = [
+        ('files', (basename, open(dest, "rb"))),
+        ('files', (text_basename, open(text_dest, "rb")))
+    ]
+    response = requests.post(dataprep_ingest_addr, headers=headers, files=files)
+    print(response.status_code)
+    os.remove(text_dest)
```

Comment:
If there are situations where we could get an exception, I would put this in a finally block so that we don't end up with lingering files when things fail.